File: app/src/main/java/com/twitli/android/twitter/data/ContentRepository.java
/*
 * Copyright (c) 2020, Zaphod Consulting BV, Christine Karman
 * This project is free software: you can redistribute it and/or modify it under the terms of
 * the Apache License, Version 2.0. You can find a copy of the license at
 * http://www.apache.org/licenses/LICENSE-2.0.
 */

package com.twitli.android.twitter.data;

import android.app.Application;

import java.util.List;

public class ContentRepository {

    private ContentDao contentDao;

    public ContentRepository(Application application) {
        AppDatabase db = AppDatabase.getDatabase(application);
        contentDao = db.contentDao();
    }

    public void addContent(String year, String datum, String text) {
        contentDao.addContent(year, datum, text);
    }

    public void addContent(String year, String text) {
        contentDao.addContent(year, text);
    }

    public Content getFirst(Integer year) {
        return contentDao.getFirst(year);
    }

    public void setDone(int id) {
        contentDao.setDone(id);
    }

    public ContentStatus getStatus(int year) {
        List<Content> available = contentDao.getAvailable(year);
        List<Content> done = contentDao.getDone(year);
        if (available.size() > 0) {
            return ContentStatus.AVAILABLE;
        } else {
            if (done.size() > 0) {
                return ContentStatus.DONE;
            } else {
                return ContentStatus.NONE;
            }
        }
    }

    public List<Content> getAvailable(int year) {
        return contentDao.getAvailable(year);
    }
}

File: app/src/main/java/com/twitli/android/twitter/data/ContentRepository.kt
/*
 * Copyright (c) 2020, Zaphod Consulting BV, Christine Karman
 * This project is free software: you can redistribute it and/or modify it under the terms of
 * the Apache License, Version 2.0. You can find a copy of the license at
 * http://www.apache.org/licenses/LICENSE-2.0.
 */
package com.twitli.android.twitter.data

import android.content.Context
import android.util.Log
import javax.inject.Inject

class ContentRepository @Inject constructor(context: Context) {
    private val contentDao: ContentDao?

    fun addContent(year: String?, datum: String?, text: String?) {

        var existingContent: Content = if (datum != null) {
            getContent(year, datum, text)
        } else {
            getContentNoDate(year, text)
        }

        if (existingContent == null) {
            contentDao!!.addContent(year, datum, text)
        }
    }

    private fun getContent(year: String?, datum: String?, text: String?): Content {
        return contentDao!!.getContentForDate(year, datum, text)
    }

    private fun getContentNoDate(year: String?, text: String?): Content {
        return contentDao!!.getContentNoDate(year, text)
    }

    fun getFirstUnused(year: String?): Content? {
        val list = contentDao!!.getFirstUnused(year)
        return if (list!!.isEmpty()) null else list[0]
    }

    fun getFirst(year: String?): Content? {
        val list = contentDao!!.getFirst(year)
        return if (list!!.isEmpty()) null else list[0]
    }

    fun getAll(year: String?): List<Content?>? {
        return contentDao!!.getAll(year)
    }

    fun setDone(id: Int) {
        contentDao!!.setDone(id)
    }

    fun getStatus(year: Int): ContentStatus {
        val available = contentDao!!.getAvailable(year)
        for (content in available!!) {
            Log.d(LOGTAG, content!!.year + " " + content.date + " " + content.text)
        }
        val done = contentDao.getDone(year)
        return if (available.size > 0) {
            ContentStatus.AVAILABLE
        } else {
            if (done!!.size > 0) {
                ContentStatus.DONE
            } else {
                ContentStatus.NONE
            }
        }
    }

    fun getAvailable(year: Int): List<Content?>? {
        return contentDao!!.getAvailable(year)
    }

    fun clear() {
        contentDao!!.clear()
    }

    companion object {
        private val LOGTAG = ContentRepository::class.java.simpleName
    }

    init {
        val db = AppDatabase.getDatabase(context!!)
        contentDao = db!!.contentDao()
    }
}
